pyccel/ast/builtins.py

```diff
@@ -299,6 +301,8 @@ def __new__(cls, arg):
 
     def __init__(self, arg):
         self._arg = arg
+        self._shape = arg.shape
+        self._rank  = len(self._shape)
```

Comment:
If we add this functionality to `PythonInt` then we should also add it to the other casting functions

Reply:
i think we should add it if we need it to prevent useless lines.

Reply:
Avoiding useless lines is good, but having different behaviour between casting functions is an easy way to confuse developers

Reply:
but we dont where we can add it now, given that the casting fine to me now.

Reply:
> but we dont know where we can add it now

What do you mean? We know which classes are casting functions. Why would we not be able to add this to one of them?

Reply:
by asking where i meant specifically, but yeah you are right about "having different behaviour between casting functions is an easy way to confuse developers"

Reply:
We should remove the [default values](https://github.com/pyccel/pyccel/blob/master/pyccel/ast/builtins.py#L142-L143) now that we calculate this systematically

Reply:
>We should remove the default values now that we calculate this systematically